tpke/src/ciphertext.rs

```diff
@@ -107,6 +132,12 @@ pub fn shared_secret_to_chacha<E: PairingEngine>(s: &E::Fqk) -> ChaCha20 {
     prf_key.write(&mut hasher).unwrap();
     let mut prf_key_32 = [0u8; 32];
     prf_key_32.clone_from_slice(hasher.finalize().as_bytes());
-    let chacha_nonce = Nonce::from_slice(b"secret nonce");
-    ChaCha20::new(Key::from_slice(&prf_key_32), chacha_nonce)
+
+    ChaCha20Poly1305::new(GenericArray::from_slice(&prf_key_32))
+}
+
+fn blinded_to_nonce<E: PairingEngine>(nonce: E::G1Affine) -> Nonce {
+    let mut nonce_bytes = Vec::new();
+    nonce.serialize_unchecked(&mut nonce_bytes).unwrap();
+    *Nonce::from_slice(&nonce_bytes[..12])
```

Comment:
Related to my last comment in #13, I'm thinking that another approach could be to hash it and then take the 12 bytes.